### PR TITLE
Denis style packed storage lemmas

### PR DESCRIPTION
**Pull request by asymmetric**
Simplifies packed storage lemmas.

### CODE REVIEW

File: src/dirty_lemmas.k.md [hunk 1]
```diff
@@ -1,2 +1,4 @@
 ```k
+rule X modInt pow32 => X
```
Comment (mhhf) [inline src/dirty_lemmas.k.md:2]: I use this rule instead:
```
rule X modInt Y => X
  requires X <Int Y
```

File: src/dirty_lemmas.k.md [hunk 2]
```diff
@@ -1,2 +1,4 @@
...
+    requires #rangeUInt(32, X)
```
Comment (mhhf) [inline src/dirty_lemmas.k.md:3]: i use this generalized rule instead:
```
rule X modInt Y => X
  requires X <Int Y
```

File: src/lemmas.k.md [hunk 3]
```diff
@@ -177,50 +168,82 @@ to match on a few different expressions for each write operation.
...
-Mask target range to zero:
+##### Zeroing
 
 ```k
 // all regions potentially non-zero
-rule (notMaxUInt112 &Int ((Z *Int pow224) +Int ((Y *Int pow112) +Int X))) => ((Z *Int pow224) +Int (Y *Int pow112))
-  requires #rangeUInt(112, X)
-  andBool #rangeUInt(112, Y)
-  andBool #rangeUInt(32, Z)
+rule X &Int ((C *Int pow224) +Int ((B *Int pow112) +Int A)) => X &Int #WordPackUInt112UInt112UInt32(A, B, C)
```
Comment (mhhf) [inline src/lemmas.k.md:175]: are the following rules needed? are they all covering the constructor case?

File: src/lemmas.k.md [hunk 4]
```diff
@@ -230,31 +253,14 @@ rule notMaxUInt112xPow112 => 115792089210356248756420345214020892766250359184300
...
-  andBool #rangeUInt(32, Z)
+rule notMaxUInt112xPow112 &Int #WordPackUInt112UInt112UInt32(A, B, C) => #WordPackUInt112UInt112UInt32(A, 0, C)
 ```
 
 Write new value:
 
 ```k
-// all regions potentially non-zero OR reserve1 is zero
-rule ((pow112 *Int Y) |Int ((Z *Int pow224) +Int X)) => ((Z *Int pow224) +Int (Y *Int pow112) +Int X)
+rule (pow112 *Int X) |Int #WordPackUInt112UInt112UInt32(A, 0, C) => #WordPackUInt112UInt112UInt32(A, X, C)
```
Comment (mhhf) [inline src/lemmas.k.md:262]: there could be code where X *Int pow112 is the case

File: src/lemmas.k.md (hunk 3 above)
Comment (asymmetric) [reply src/lemmas.k.md:182]: Maybe they aren't, you're right. After the `SLOAD` we have `#WordPack` on the stack, and then it's `AND`ed with a `maxUInt`.

File: src/lemmas.k.md (hunk 4 above)
Comment (asymmetric) [reply src/lemmas.k.md:269]: True. We should normalize this too I guess.

File: src/dirty_lemmas.k.md (hunk 2 above)
Comment (asymmetric) [reply src/dirty_lemmas.k.md]: This should probably go to `rules.k.tmpl`, right?

File: src/lemmas.k.md (hunk 4 above)
Comment (asymmetric) [reply src/lemmas.k.md:308]: I'm  going to leave it as is, if it breaks something we'll add the lemma with the terms reversed.

File: src/lemmas.k.md [hunk 5]
```diff
@@ -212,6 +219,12 @@ rule takeWordStack(20, #asByteStack(X)) =>
 
 ### Packed Storage { `uint32` `uint112` `uint112` }
 
+Define the symbolic term representing packed storage:
```
Comment (asymmetric) [inline src/lemmas.k.md:222]: Not sure if this makes sense, could probably be worded better.

**Comment by asymmetric** [comment]
- [Part 1](https://buildbot.dapp.ci/#/builders/12/builds/628/steps/3/logs/stdio), with the weird `BADGAS`
- [Part 2](https://buildbot.dapp.ci/#/builders/12/builds/630), where I manually removed that spec from `accept` and hit rebuild

This seems good to go.

**Pull request merged**